File: REST/DMS.Domain/Entities/Document.cs
namespace DMS.Domain.Entities;

using DMS.Domain.ValueObjects;

public class Document
{
    public int Id { get; set; }
    public string? Content { get; set; }
    public DateTime UploadDateTime { get; set; }
    public DateTime ModificationDateTime { get; set; }
    public string Path { get; set; }
    public List<Tag> Tags { get; set; }
    public FileType DocumentType { get; set; }
    public ProcessingStatus Status { get; set; }
}

File: frontend/src/types/Document.ts
import { Tag } from "./Tag";

export type UploadDocumentDto = {
    title: string;
    content: string;
    tags: Tag[];
    fileType: string;
  }

export type UpdateDocumentDto = {
    id: string;
    title: string;
    tags: Tag[];
}

export type Document = {
    id: string;
    title: string;
    content: string;
    tags: Tag[];
    modificationDateTime: string;
    uploadDateTime: string;
    status: DocumentStatus;
    fileExtension: string;
  }

  export type DocumentContentDto = {
    id: string;
    content: string;
  }

  export enum DocumentStatus {
    Pending,
    NotStarted,
    Finished,
    Failed
  }